scripts/stablex/setup_thegraph_data.js

```diff
@@ -0,0 +1,118 @@
+const { solutionSubmissionParams, basicTrade } = require("../../test/resources/examples")
+
+const { makeDeposits, placeOrders, createMintableToken, mintTokens } = require("../../test/stablex/stablex_utils")
+const {
+  closeAuction,
+  getOwl,
+  getBatchExchange,
+  mintOwlForUsers,
+  setAllowances,
+  addTokens,
+  deleteOrders,
+  submitSolution,
+  getBatchId,
+} = require("../../scripts/stablex/utilities.js")
+
+const { toETH } = require("../../test/resources/math")
+const { waitForNSeconds } = require("../../test/utilities")
+
+module.exports = async function(callback) {
+  try {
+    const batchExchange = await getBatchExchange(artifacts)
+    const owl = await getOwl(artifacts)
+
+    // Prepare user accounts
+    const accounts = await web3.eth.getAccounts()
+    const [user1Address, user2Address] = accounts
+    const usersAddresses = [user1Address, user2Address]
+    const solverAddress = user1Address
+
+    // Get current batch id
+    let batchId = await getBatchId(batchExchange)
+    console.log("Current batch id: ", batchId)
+
+    // Set user1 as minter of OWL
+    await owl.setMinter(user1Address)
+
+    // Mint OWL for every user
+    const amount = web3.utils.toWei("3000")
+    await mintOwlForUsers({ users: usersAddresses, amount, owl })
+
+    // Create 1 token
+    const token1Instance = await createMintableToken(artifacts)
+    const tokensInstances = [token1Instance]
+
+    // Set allowances for OWL and the tokens
+    await setAllowances({
+      users: usersAddresses,
+      amount,
+      batchExchange,
+      tokens: tokensInstances.concat([owl]),
+    })
+
+    // List the tokens in the exchange
+    const tokenAddresses = [token1Instance.address]
+    const [token1] = await addTokens({ tokenAddresses, account: user1Address, batchExchange, owl })
+
+    // Mint tokens for every user
+    await mintTokens({
+      tokens: tokensInstances,
+      users: [user1Address, user2Address],
+      amount,
+    })
+
+    // Make deposits, place orders and close auction [aka runAuctionScenario(basicTrade)]
+    await makeDeposits(batchExchange, usersAddresses, basicTrade.deposits)
+
+    // Place orders
+    let orderIds = await placeOrders(batchExchange, usersAddresses, basicTrade.orders, batchId + 1)
+
+    // Request withdraw
+    console.log("Request withdraw for user %s, token %s (%d)", user1Address, token1.address, token1.id)
+    await batchExchange.requestWithdraw(token1.address, 5, { from: user1Address })
+
+    // Close the auction
+    console.log("Close auction so we can withdraw the tokens")
+    await closeAuction(batchExchange, web3)
+
+    // Withdraw tokens
+    console.log("Withdraw for user %s, token %s (%d)", user1Address, token1.address, token1.id)
+    await batchExchange.withdraw(user1Address, token1.address, { from: user1Address })
+
+    // Submit solution
+    await submitSolution({
```

Comment:
Did you want to use an example with a solution and a better solution to get revert events?

Reply:
It would be nice! it was in my plans. I would love to do that in another PR